### PR TITLE
MySQL collation support

### PR DESCRIPTION
**Pull request by BillyYccc**

### CODE REVIEW

**Comment by BillyYccc** [comment]
With collation support we can later support MySQL charset easily and make charset configuration a shortcut for configuring a correlative collation.

**Comment by BillyYccc** [comment]
We also switch the default collation from `utf8_general_ci` to `utf8mb4_general_ci`.

**Comment by vietj** [comment]
can we make the mysqlcollation object simply a String ?

File: vertx-mysql-client/src/test/java/io/vertx/mysqlclient/MySQLQueryTest.java [hunk 1]
```diff
@@ -38,6 +39,133 @@ public void teardown(TestContext ctx) {
     vertx.close(ctx.asyncAssertSuccess());
   }
 
+  @Test
+  public void testTableCollation(TestContext ctx) {
+    MySQLConnection.connect(vertx, options, ctx.asyncAssertSuccess(conn -> {
+      conn.query("CREATE TEMPORARY TABLE chinese_city (\n" +
+        "\tid INTEGER,\n" +
+        "\tcity_name VARCHAR(20)) COLLATE gbk_chinese_ci;\n" +
```
Comment (vietj) [inline vertx-mysql-client/src/test/java/io/vertx/mysqlclient/MySQLQueryTest.java]: we need to use unicode escape for the Chinese chars,  i.e `\uXXX` so they work with any file editor

Comment (BillyYccc) [reply vertx-mysql-client/src/test/java/io/vertx/mysqlclient/MySQLQueryTest.java]: Fixed in https://github.com/eclipse-vertx/vertx-sql-client/pull/352/commits/1f91501eb1ccb871e05f41adef1cb6b59f094aec.

**Comment by BillyYccc** [comment]
> can we make the mysqlcollation object simply a String ?

In that way we may need additional check if the given collation name is illegal or not.

**Comment by BillyYccc** [comment]
I will rebase on the `3.8` branch to resolve the conflicts soon.

**Comment by BillyYccc** [comment]
I have make the branch rebased on the `3.8` brach.

**Comment by vietj** [comment]
we need to use a `String` for `MySQLCollation` instead of an enum which is a too restrictive choice

**Comment by BillyYccc** [comment]
updated with https://github.com/eclipse-vertx/vertx-sql-client/pull/352/commits/df24dd118870212d16e785e9d8cf073cc5f7877c, I will add  charset shortcut configuration and documentation if you are fine with these changes.


> we need to use a `String` for `MySQLCollation` instead of an enum which is a too restrictive choice

**Comment by BillyYccc** [comment]
@vietj Can you please make another review for this? I wish to make this feature added in the 3.8.0 release.

**Comment by BillyYccc** [comment]
Thanks! I will add support for charset and documentation soon.

**Pull request merged**